app/src/main/java/com/quizarena/menu/MainMenuActivity.kt

```diff
@@ -46,20 +47,32 @@ class MainMenuActivity : AppCompatActivity() {
             // the activity was opened by a sharing link
             // go to the needed activity
             val session = SessionSharing().getSessionFromLink(this, intent.data)
-            val detailViewIntent = Intent(this@MainMenuActivity, SessionDetailActivity::class.java)
-            detailViewIntent.putExtra(getString(R.string.intent_extra_session_clicked), session)
-            startActivity(detailViewIntent)
-
-            // override the intent action for going back in the other activity
-            this.intent.action = Intent.ACTION_MAIN
+            startSessionDetailActivity(session)
 
         } else {
-            // activity was started manually, not by a link
+            val sessionID = intent.getStringExtra("session")
+            if(sessionID != null){
```

Comment:
Ich komme hier in eine Endlosschleife. Wenn ich auf Zurück klicke, dann wird der gleiche Intent aufgerufen, der diesen Parameter gesetzt hat. Ich werde dann also direkt wieder zur DetailView geleitet. Das Extra des Intent muss auf null gesetzt werden. Ich denk, das sollte es dann lösen.

gleicher Grund warum ich die intent.action in Zeile 74 setze

Reply:
Ist geändert